File: Domain/Entities/PannaPramukh.cs
namespace Nirvachak_AI.Domain.Entities;

public class PannaPramukh
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string? Address { get; set; }
    public int BoothNumber { get; set; }
    public string PannaNumber { get; set; } = string.Empty;    // Panna page (e.g. "3A")
    public int ConstituencyId { get; set; }
    public Constituency? Constituency { get; set; }
    public int TotalVotersAssigned { get; set; }
    public int VotersContacted { get; set; }
    public bool IsActive { get; set; } = true;
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: mobile/src/api/pannaPramukh.ts
import apiClient from './client';

export interface PannaPramukhItem {
  id: number;
  name: string;
  phone: string;
  boothNumber: number;
  pannaNumber: string;
  totalVotersAssigned: number;
  votersContacted: number;
  contactPercent: number;
  isActive: boolean;
  notes?: string;
}

export const getPannaPramukhs = async (booth?: number): Promise<PannaPramukhItem[]> => {
  const { data } = await apiClient.get<PannaPramukhItem[]>('/pannapramukh', {
    params: booth ? { booth } : {},
  });
  return data;
};

export const createPannaPramukh = async (req: {
  name: string; phone: string; email?: string; address?: string;
  boothNumber: number; pannaNumber: string;
  totalVotersAssigned: number; notes?: string;
}): Promise<void> => {
  await apiClient.post('/pannapramukh', req);
};

export const updatePannaContact = async (id: number, votersContacted: number): Promise<void> => {
  await apiClient.put(`/pannapramukh/${id}/contact`, { id, votersContacted });
};
